### ISSUE TITLE
Allow `as_html` to produce compact HTML.

### ISSUE THREAD

**Issue by RobertDober** [pragdave/earmark#329]
The default AST->HTML Transform as foreseen for 1.5 renders like the following

```elixir
iex(54)> as_html("Elixir **is** best")
{:ok, "<p>\n  Elixir \n  <strong>\n    is\n  </strong>\n   best\n</p>\n", []}
``` 

I had not given that much thought yet but I reckon that this pretty printing is not really what we might want for ex_doc.

For that reason I think that I will implement

```elixir
iex(54)> as_html("Elixir **is** best", pretty: false)
{:ok, "<p>\nElixir <strong>is</strong>best\n</p>\n", []}
```

@josevalim would you prefer to use the `pretty: false` inside *ExDoc*?

**Comment by RobertDober** [comment]
**N.B.** It is also a bug as right now we add whitespace between elements where we should not, e.g.
between two links

**Comment by corybuecker** [comment]
Hi @RobertDober, I ran into this too and created a fork. I'd like to contribute it back in, but wanted to check my approach first.

I have added a new key to the [options passed into the `Transform` module](https://github.com/pragdave/earmark/blob/master/lib/earmark/transform.ex#L33). If `pretty` is false, it will override the indentation, setting it to 0, and prevent the addition of newline characters to the lists generated by `_to_html`. Does that sound about right?

**Comment by RobertDober** [comment]
Hmm I'am afraid that we still want do add some newlines.
My idea was to define a list of tags, `<p>, <table>, <thead>...` which will be rendered like so
```html
<p>
#{content}
</p>   
```
but as a matter of fact the output of 1.4.3 should be used as a guideline, but it is not perfect either

That said, I appreciate a PR but am extremely busy with my paid job right now and have no idea when I can have a look at it :sob:

**Comment by chazsconi** [comment]
I think a list of tags for pretty printing is required when rendering because having `\n` output between all tags currently causes `Elixir is b**e**st` to be rendered like this when transforming the ast:

```
<p>
  Elixir is b
  <strong>
    e
  </strong>
  st
</p>"
```
Which will show with unwanted whitespace in a browser: i.e. "Elixir is b **e** st"

**Comment by RobertDober** [comment]
@chazsconi of course, formatting tags like `em, strong..` should be rendered _inline_

**Comment by tmjoen** [comment]
Did this get added to 1.4.5? Couldn't find anything in the changelog but my tests were suddenly failing:

```
     Earmark.to_html!("**Some** text here.")

     1.4.5:  "<p>\n  <strong>\n    Some\n  </strong>\n   text here.\n</p>\n"
     1.4.4: "<p><strong>Some</strong> text here.</p>\n"
```

**Comment by RobertDober** [comment]
this is strange, I do not recall having done any work on this...

This ticket will stabilize the format of the rendered AST

**Comment by RobertDober** [comment]
This is fixed with #371
